create.ps1

```diff
@@ -67,80 +50,115 @@ function Get-ADPAccessToken {
+        } catch {
+            $httpErrorObj.FriendlyMessage = $httpErrorObj.ErrorDetails
         }
         Write-Output $httpErrorObj
     }
 }
 #endregion
 
+
 try {
-    # Begin
-    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
-    $certificate = [System.Security.Cryptography.X509Certificates.X509Certificate2]::new($config.CertificatePath, $config.CertificatePassword)
-    $accessToken = Get-ADPAccessToken -ClientID $($config.ClientID) -ClientSecret $($config.ClientSecret) -Certificate $certificate
-    
-    $headers = @{
-        "Authorization" = "Bearer $($accessToken.access_token)"
-    }
+    # Initial Assignments
+    $outputContext.AccountReference = 'Currently not available'
 
-    Write-Verbose "Verify if ADPWorkforce account for: [$($p.DisplayName)] exists"
- 
-    $splatParams = @{
-        Uri         = "$($config.BaseUrl)/hr/v2/worker-demographics/$($p.Custom.AssociateOID)"
-        Method      = 'GET'
-        Headers     = $headers
-        Certificate = $certificate
-    }
-    $responseGetUser = Invoke-RestMethod @splatParams
-
-    if ($responseGetUser.Workers[0].associateOID -eq $($p.Custom.AssociateOID)) {
-        # If the E-mail address in HelloID matches with the E-mail address in ADPWorkforce -> Correlate
-        Write-Verbose "Verifying if the E-mail address for: [$($p.DisplayName)] must be updated"
-        if ($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri -eq $account.workerEmail) {
-            $action = 'Correlate'
-            # If the E-mail address in HelloID differs from the E-mail address in ADPWorkforce -> Correlate-Update
+    # Validate correlation configuration
+    if ($actionContext.CorrelationConfiguration.Enabled) {
+        $correlationField = $actionContext.CorrelationConfiguration.accountField
+        $correlationValue = $actionContext.CorrelationConfiguration.accountFieldValue
+
+        if ([string]::IsNullOrEmpty($($correlationField))) {
+            throw 'Correlation is enabled but not configured correctly'
         }
-        elseif ($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri -ne $account.workerEmail) {
-            $action = 'Correlate-Update'
+        if ([string]::IsNullOrEmpty($($correlationValue))) {
+            throw 'Correlation is enabled but [accountFieldValue] is empty. Please make sure it is correctly mapped'
         }
-    }
 
+        # Verify if a user must be either [created] or just [correlated]
+        [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
+        $certificate = [System.Security.Cryptography.X509Certificates.X509Certificate2]::new($actionContext.Configuration.CertificatePath, $actionContext.Configuration.CertificatePassword)
```

Comment:
In the source connector, there's code to load a certificate directly from HelloID. Since the source connector is a requirement for this connector, it would be nice to include the same functionality within the update connector. Also ensures consistency.